From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Victor Toso <me@victortoso.com>
Date: Mon, 11 Jun 2018 12:14:30 +0200
Subject: [PATCH] Revert "channel-usbredir: Fix crash on channel-up"

This reverts commit 291f3e4419e6fb4077ae43a5e09eb1c37b9dd729.

Follow up patch should address rhbz#1399838 mentioned in commit above
in a different way.

Major reason to revert is that the SpiceUsbDeviceManager object is
kept in SpiceSession as an easy way to share it between different
SpiceUsbredirChannel while SpiceSession itself does not use it. This
causes problems on migration as we start a new session with a new
SpiceUsbDeviceManager object while the previous one still exists.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1558043
Signed-off-by: Victor Toso <victortoso@redhat.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 src/spice-session.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/spice-session.c b/src/spice-session.c
index a729cc3..094dc41 100644
--- a/src/spice-session.c
+++ b/src/spice-session.c
@@ -281,7 +281,6 @@ static void spice_session_init(SpiceSession *session)
 {
     SpiceSessionPrivate *s;
     gchar *channels;
-    GError *err = NULL;
 
     SPICE_DEBUG("New session (compiled from package " PACKAGE_STRING ")");
     s = session->priv = SPICE_SESSION_GET_PRIVATE(session);
@@ -294,12 +293,6 @@ static void spice_session_init(SpiceSession *session)
     s->images = cache_image_new((GDestroyNotify)pixman_image_unref);
     s->glz_window = glz_decoder_window_new();
     update_proxy(session, NULL);
-
-    s->usb_manager = spice_usb_device_manager_get(session, &err);
-    if (err != NULL) {
-        SPICE_DEBUG("Could not initialize SpiceUsbDeviceManager - %s", err->message);
-        g_clear_error(&err);
-    }
 }
 
 static void
